docs: update Docker image release step (#375)
jenkins-infra/release: docs/releases.md

Adjust state and `Released As` of [Jira issues](https://issues.jenkins.io/) fixed in the release (see the [changelog](https://www.jenkins.io/changelog-stable) for issue links)

### Publish Docker image

> ⚠️ **Requires write access to the repository jenkinsci/docker** ⚠️

Reviewer: In case a hyperlink helps a reader understand what is being said...
```suggestion
⚠️ **Requires write access to the [jenkinsci/docker repository](https://github.com/jenkinsci/docker) ** ⚠️
```

> 💡 **Access to the network trusted.ci.jenkins.io is could help in case of error but is not mandatory** 💡 

Reviewer: ```suggestion
💡 **Access to the network containing trusted.ci.jenkins.io could help in case of error but is not mandatory** 💡 
```

> Create a new tag in the repository jenkinsci/docker which value is the released core version.

Reviewer: ```suggestion
Create a new tag in the [jenkinsci/docker repository](https://github.com/jenkinsci/docker) with the value of the released core version.
```
